c/src/ml-api-service-remote.c

```diff
@@ -563,81 +565,77 @@ ml_service_remote_release_internal (ml_service_s * mls)
     nns_edge_release_handle (mlrs->edge_h);
-
-    /* Wait some time until release the edge handle. */
-    g_usleep (1000000);
   }
 
   g_free (mlrs->path);
+  ml_option_destroy (mlrs->info);
   g_free (mlrs);
 
   return ML_ERROR_NONE;
 }
 
+/**
+ * @brief Internal function to set path in ml-service remote handle.
+ */
+static int
+ml_service_remote_set_path (ml_service_s * mls)
+{
+  _ml_remote_service_s *mlrs = (_ml_remote_service_s *) mls->priv;
+  gchar *_path = NULL;
+
+  if (ML_ERROR_NONE == ml_option_get (mlrs->info, "path", (void **) (&_path))) {
+    if (!g_file_test (_path, G_FILE_TEST_IS_DIR)) {
+      _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
+          "The given param, dir path = \"%s\" is invalid or the dir is not found or accessible.",
+          _path);
+    }
+    if (g_access (_path, W_OK) != 0) {
+      _ml_error_report_return (ML_ERROR_PERMISSION_DENIED,
+          "Write permission denied, path: %s", _path);
+    }
+  }
+  mlrs->path = g_strdup (_path);
+
+  return ML_ERROR_NONE;
+}
+
 /**
  * @brief Creates ml-service handle with given ml-option handle.
  */
 int
-ml_service_remote_create (ml_option_h option, ml_service_event_cb cb,
-    void *user_data, ml_service_h * handle)
+ml_service_remote_create (ml_service_h handle, ml_option_h option)
 {
   ml_service_s *mls;
   _ml_remote_service_s *remote_s = NULL;
   edge_info_s *edge_info = NULL;
   int ret = ML_ERROR_NONE;
-  gchar *_path = NULL;
 
   check_feature_state (ML_FEATURE_SERVICE);
   check_feature_state (ML_FEATURE_INFERENCE);
 
-  if (!option) {
-    _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-        "The parameter, 'option' is NULL. It should be a valid ml_option_h, which should be created by ml_option_create().");
-  }
 
   if (!handle) {
     _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
         "The parameter, 'handle' (ml_service_h), is NULL. It should be a valid ml_service_h.");
   }
-
-  if (ML_ERROR_NONE == ml_option_get (option, "path", (void **) (&_path))) {
-    if (!g_file_test (_path, G_FILE_TEST_IS_DIR)) {
-      _ml_error_report_return (ML_ERROR_INVALID_PARAMETER,
-          "The given param, dir path = \"%s\" is invalid or the dir is not found or accessible.",
-          _path);
-    }
-    if (g_access (_path, W_OK) != 0) {
-      _ml_error_report_return (ML_ERROR_PERMISSION_DENIED,
-          "Write permission denied, path: %s", _path);
-    }
-  }
-
-  mls = _ml_service_create_internal (ML_SERVICE_TYPE_REMOTE);
-  if (mls == NULL) {
-    _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
-        "Failed to allocate memory for the service handle. Out of memory?");
-  }
+  mls = (ml_service_s *) handle;
 
   mls->priv = remote_s = g_try_new0 (_ml_remote_service_s, 1);
   if (remote_s == NULL) {
-    _ml_service_destroy_internal (mls);
     _ml_error_report_return (ML_ERROR_OUT_OF_MEMORY,
         "Failed to allocate memory for the service handle's private data. Out of memory?");
   }
+  remote_s->info = option;
```

Comment:
If we just use `ml_option_h option`, can I know what happens if the option changes or release outside of the service after?
If unintended changes can occur, how about add a function like `ml_option_copy or clone()`?

Reply:
As you mentioned, it is dangerous to copy only the pointer of the ml-option.
And instead of copying the whole ml-option, it was changed to save only the necessary information as an property of ml-service.